### DISCUSSION TITLE
How to force fresh data (no cache) from query

### DISCUSSION THREAD

**Discussion by Eduard-Hasa** [TanStack/query#4666] [Q&A] [answered]
Is there a way to tell a specific query call to bypass cache? I seen the docs on cache but a little confused how I could do it without having to worry about invalidating the cache first. Simply tell a query to get new fresh data from api.

**Comment by TkDodo** [comment, marked as answer]
caching is performed on a time basis. Depending on the set `staleTime`, react-query will fetch new data or only read from the cache. To "force" a refetch, you have to imperatively call a method that refetches, e.g. `queryClient.invalidateQueries` or `queryClient.refetchQueries`

**Reply by Eduard-Hasa** [reply]
Thank you for the answer and this amazing project.

**Comment by mizanmahi** [comment]
Can i you give an example code snippet? how can I remove caching for a API call.

**Comment by samirarezai** [comment]
```
 const { data, isLoading } = useQuery({
    queryKey: ['sample'],
    queryFn: calcConfigAPI.getTestTemplate,
    gcTime: 0,
  })
```

  You should use `gcTime: 0,` to inactive cache data

**Reply by Akshay090** [reply]
i think using gcTime: 0 for this case is a anti pattern, reason being 
gcTime: 0 doesn't work when we have useSuspenseQuery as it can get the page in infinitely suspended state

**Reply by samirarezai** [reply]
Oh didnt know that, thanks
